controllers/bpfman-agent/application-program.go

```diff
@@ -0,0 +1,322 @@
+				fexitProgram := bpfmaniov1alpha1.FexitProgram{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: a.Name + "-fexit",
+					},
+					Spec: bpfmaniov1alpha1.FexitProgramSpec{
+						FexitProgramInfo: *p.Fexit,
+						BpfAppCommon:     a.Spec.BpfAppCommon,
+					},
+				}
+				rec := &FexitProgramReconciler{
+					ReconcilerCommon:    r.ReconcilerCommon,
+					currentFexitProgram: &fexitProgram,
+					ourNode:             r.ourNode,
+				}
+				rec.appOwner = &a
+				fexitObjects := []client.Object{&fexitProgram}
+				// Reconcile FexitProgram.
+				complete, res, err = r.reconcileCommon(ctx, rec, fexitObjects)
+
+				r.showPrograms(ctx, rec)
+
+			case bpfmaniov1alpha1.ProgTypeKprobe,
+				bpfmaniov1alpha1.ProgTypeKretprobe:
+				kprobeProgram := bpfmaniov1alpha1.KprobeProgram{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: a.Name + "-kprobe",
+					},
+					Spec: bpfmaniov1alpha1.KprobeProgramSpec{
+						KprobeProgramInfo: *p.Kprobe,
+						BpfAppCommon:      a.Spec.BpfAppCommon,
+					},
+				}
+				rec := &KprobeProgramReconciler{
+					ReconcilerCommon:     r.ReconcilerCommon,
+					currentKprobeProgram: &kprobeProgram,
+					ourNode:              r.ourNode,
+				}
+				rec.appOwner = &a
+				kprobeObjects := []client.Object{&kprobeProgram}
+				// Reconcile KprobeProgram or KpretprobeProgram.
+				complete, res, err = r.reconcileCommon(ctx, rec, kprobeObjects)
+
+				r.showPrograms(ctx, rec)
+
+			case bpfmaniov1alpha1.ProgTypeUprobe,
+				bpfmaniov1alpha1.ProgTypeUretprobe:
+				uprobeProgram := bpfmaniov1alpha1.UprobeProgram{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: a.Name + "-uprobe",
+					},
+					Spec: bpfmaniov1alpha1.UprobeProgramSpec{
+						UprobeProgramInfo: *p.Uprobe,
+						BpfAppCommon:      a.Spec.BpfAppCommon,
+					},
+				}
+				rec := &UprobeProgramReconciler{
+					ReconcilerCommon:     r.ReconcilerCommon,
+					currentUprobeProgram: &uprobeProgram,
+					ourNode:              r.ourNode,
+				}
+				rec.appOwner = &a
+				uprobeObjects := []client.Object{&uprobeProgram}
+				// Reconcile UprobeProgram or UpretprobeProgram.
+				complete, res, err = r.reconcileCommon(ctx, rec, uprobeObjects)
+
+				r.showPrograms(ctx, rec)
+
+			case bpfmaniov1alpha1.ProgTypeTracepoint:
+				tracepointProgram := bpfmaniov1alpha1.TracepointProgram{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: a.Name + "-tracepoint",
+					},
+					Spec: bpfmaniov1alpha1.TracepointProgramSpec{
+						TracepointProgramInfo: *p.Tracepoint,
+						BpfAppCommon:          a.Spec.BpfAppCommon,
+					},
+				}
+				rec := &TracepointProgramReconciler{
+					ReconcilerCommon:         r.ReconcilerCommon,
+					currentTracepointProgram: &tracepointProgram,
+					ourNode:                  r.ourNode,
+				}
+				rec.appOwner = &a
+				tracepointObjects := []client.Object{&tracepointProgram}
+				// Reconcile TracepointProgram.
+				complete, res, err = r.reconcileCommon(ctx, rec, tracepointObjects)
+
+				r.showPrograms(ctx, rec)
```

Comment:
I think we can remove this... if we need more logging please add it to the subcalls of `r.reconcileCommon`

Reply:
removed debug code